### ISSUE TITLE
schedule.wpi is 1h too early

### ISSUE THREAD

**Issue by decapeter2** [uugear/Witty-Pi-2#10]
Hello :)
i've generated a 9h ON state per day with the uugear tool page and the script is running BUT :

- it should ON at 08:30 and OFF at 19:00
- it turn ON at 07:30 and OFF at 18:00 (1h too early)
- the pi and the witty are both at the right time sync from internet
- the pi conf is Paris timezone +1

Is it a bug or i did something wrong?

**Comment by uugear** [comment]
Please show us the schedule script you are using. Currently there is nothing we can look into.
Meanwhile have you run the simulation with [our web app](http://www.uugear.com/app/wittypi-scriptgen/)?

**Comment by decapeter2** [comment]
yes , i've generated the script from your online tool page and tested it too with correct time shown.
Do you still want the script?

**Comment by uugear** [comment]
Have you run "sudo dpkg-reconfigure tzdata" and configured the timezone properly? When you run "sudo ./wittyPi.sh", what system time and RTC time did you see? Are they matched?

**Comment by decapeter2** [comment]
when i run "sudo ./wittyPi.sh" both RTC are sync (only 1sec diff)

About "sudo dpkg-reconfigure tzdata", if it's the command asked during the install process, i said yes to run it.

The time zone was configured lot time ago with the noob interface "configuration" pannel and "europe paris" was selected.

**Comment by decapeter2** [comment]
for information, even on the command line "sudo ./wittyPi.sh", the "5. Schedule next startup" is also 1h too early (same as the API i used before)

**Comment by Joehc** [comment]
What is you begin and end date? I had the same problem, but it was due to the summer/winter time.

**Comment by decapeter2** [comment]
Here is my script: 
BEGIN 2015-08-03 08:30:00
END   2025-07-31 19:00:00
ON    H10 M30
OFF   H13 M30
ON    H10 M30
OFF   H13 M30
ON    H10 M30
OFF   H13 M30
ON    H10 M30
OFF   H13 M30
ON    H10 M30
OFF   D2 H13 M30

**Comment by uugear** [comment]
Joehc may be right.

What if you change the first line to "BEGIN 2017-03-01 08:30:00" ?

**Comment by decapeter2** [comment]
why not try that but what will happend on the next time season changing? will it be 1h too late? the witty pi card isnt able to use season time? :/

**Comment by uugear** [comment]
More specifically, it is the RTC chip doesn't support daylight saving time. As far as I know, all RTC chips uses GMT internally, and do not support daylight saving time.

If we implement daylight saving time in our own software, the workload will be very big, as different countries have different daylight saving times... So currently we don't do it.

If your country has daylight saving time, you should prepare two schedule scripts: one for summer and one for winter. You can switch these two scripts manually or use a program to automatically switch them.

In future version of the software we will allow user to input their local daylight saving time parameters, and shift the time accordingly.

**Comment by decapeter2** [comment]
thank you for explaination. 
Yes we have daylight saving time in france. Unfortunately i can't change the script time over all my pi park becose they are in my customers company everywhere in the country and i cant imagine to change all of them one by one with the remote desktop 2 times per year.
If the witty pi could be "intelligent" by using internet time like my Windows do or my Android, i could use witty but actually, it was a test and i think i will stop here. It's too complicated to use it. Plus, i have a problem to find a "pro" box for the pi/witty (the transparent ones on uugear website are too "light").
Thank you for your answers and good luck.

**Comment by uugear** [comment]
You are welcome. Just two more comments:

You don't have to access the RPi again to switch the script. It can be done with a shell script at startup, who automatically checks the date and replace a file from backup directory.

Witty Pi does use Internet time. If your Raspberry Pi has Internet access, the daylight saving time shouldn't be a problem. Because at startup Witty Pi's software will automatically apply the network time to both Raspberry Pi and Witty Pi. If the network time is a daylight saving time, your Witty Pi and system will get the same.

**Comment by decapeter2** [comment]
hooo, yes they are all connected to internet then daylight saving time should be ok :)
thank you for the advice !
